apps/rush-lib/src/pluginFramework/PluginManager.ts

```diff
@@ -0,0 +1,206 @@
+  private _installedAutoinstallerNames: Set<string>;
+  private _loadedPluginNames: Set<string> = new Set<string>();
+
+  private _error: Error | undefined;
+
+  public constructor(options: IPluginManagerOptions) {
+    this._terminal = options.terminal;
+    this._rushConfiguration = options.rushConfiguration;
+    this._rushSession = options.rushSession;
+
+    this._installedAutoinstallerNames = new Set<string>();
+
+    const defaultPluginConfigurations: IDefaultRushPluginConfiguration[] = [];
+    this._defaultPluginLoaders = defaultPluginConfigurations.map((pluginConfiguration) => {
+      return new DefaultPluginLoader({
+        pluginConfiguration,
+        rushConfiguration: this._rushConfiguration,
+        terminal: this._terminal
+      });
+    });
+
+    this._remotePluginLoaders = (
+      this._rushConfiguration?.rushPluginsConfiguration.configuration.plugins ?? []
+    ).map((pluginConfiguration) => {
+      return new RemotePluginLoader({
+        pluginConfiguration,
+        rushConfiguration: this._rushConfiguration,
+        terminal: this._terminal
+      });
+    });
+  }
+
+  public get error(): Error | undefined {
+    return this._error;
+  }
+
+  public addDefaultPluginConfigurations(
+    defaultPluginConfigurations: IDefaultRushPluginConfiguration[]
+  ): void {
+    for (const pluginConfiguration of defaultPluginConfigurations) {
+      this._defaultPluginLoaders.push(
+        new DefaultPluginLoader({
+          pluginConfiguration,
+          rushConfiguration: this._rushConfiguration,
+          terminal: this._terminal
+        })
+      );
+    }
+  }
+
+  public async updateAsync(): Promise<void> {
+    await this._preparePluginAutoinstallersAsync(this._remotePluginLoaders);
+    const preparedAutoinstallerNames: Set<string> = new Set<string>();
+    for (const { autoinstaller } of this._remotePluginLoaders) {
+      const storePath: string = RemotePluginLoader.getPluginAutoinstallerStorePath(autoinstaller);
+      if (!preparedAutoinstallerNames.has(autoinstaller.name)) {
+        FileSystem.ensureEmptyFolder(storePath);
+        preparedAutoinstallerNames.add(autoinstaller.name);
+      }
+    }
+    for (const pluginLoader of this._remotePluginLoaders) {
+      pluginLoader.update();
+    }
+  }
+
+  public async reinitializeAllPluginsForCommandAsync(commandName: string): Promise<void> {
+    this._error = undefined;
+    await this.tryInitializeUnassociatedPluginsAsync();
+    await this.tryInitializeAssociatedCommandPluginsAsync(commandName);
+  }
+
+  public async _preparePluginAutoinstallersAsync(pluginLoaders: RemotePluginLoader[]): Promise<void> {
+    for (const { autoinstaller } of pluginLoaders) {
+      if (!this._installedAutoinstallerNames.has(autoinstaller.name)) {
+        await autoinstaller.prepareAsync();
+        this._installedAutoinstallerNames.add(autoinstaller.name);
+      }
+    }
+  }
+
+  public async tryInitializeUnassociatedPluginsAsync(): Promise<void> {
```

Comment:
The idea is plugin errors should be ignored when user runs `update`, `init-autoinstaller` or `update-autoinstaller`, because those 3 commands are used to fix plugin configurations. This is what `try` represents for.

Reply:
It does attempt to perform an action which may fail.

@dmichon-msft  I think the confusion is that Microsoft has a broad convention that `tryDoSomeAction()` will return `false` if the action cannot be performed.

My fix is to rename it to `attemptInitializeUnassociatedPluginsAsync()`.